packages/runtime/test/src/interpreter/pooling-inner-transactions-test.ts

```diff
@@ -0,0 +1,375 @@
+                .get(appAccAddr)?.balance()
+                , BigInt(balance) - 3000n);
+            assert.equal(interpreter.runtime.ctx.state.accounts
+                .get(johnAcc.address)?.balance()
+                , 0n);
+        });
+
+        it("Should succeed: when txn fee is covered in group txns", function () {
+            assert.equal(interpreter.runtime.ctx.state.accounts
+                .get(johnAcc.address)?.balance()
+                , 0n);
+
+            const prog = `
+				itxn_begin
+				int pay
+				itxn_field TypeEnum
+				txn Applications 2
+				app_params_get AppAddress
+				assert
+				itxn_field Receiver
+				int 1000
+				itxn_field Amount
+				int 1000
+				itxn_field Fee
+				itxn_next
+				int pay
+				itxn_field TypeEnum
+				txn Sender
+				itxn_field Receiver
+				int 1000
+				itxn_field Amount
+				int 1000
+				itxn_field Fee
+				itxn_submit
+				int 1
+				return
+				`;
+            assert.doesNotThrow(() => executeTEAL(prog));
+
+            // verify deducted fee
+            assert.equal(interpreter.runtime.ctx.state.accounts
+                .get(appAccAddr)?.balance()
+                , BigInt(balance) - 4000n);
+            assert.equal(interpreter.runtime.ctx.state.accounts
+                .get(johnAcc.address)?.balance()
+                , 0n);
+        });
+
+        it("Should fail: when txn fee is not covered in group txns", function () {
+            assert.equal(interpreter.runtime.ctx.state.accounts
+                .get(johnAcc.address)?.balance()
+                , 0n);
+
+            const prog = `
+				itxn_begin
+				int pay
+				itxn_field TypeEnum
+				txn Applications 2
+				app_params_get AppAddress
+				assert
+				itxn_field Receiver
+				int 1000
+				itxn_field Amount
+				int 0
+				itxn_field Fee
+				itxn_next
+				int pay
+				itxn_field TypeEnum
+				txn Sender
+				itxn_field Receiver
+				int 1000
+				itxn_field Amount
+				// txn fee of 1000 is covered but 2000 is required
+				int 1000
+				itxn_field Fee
+				itxn_submit
+				int 1
+				return
+				`;
+
+            expectRuntimeError(
+                () => executeTEAL(prog),
+                RUNTIME_ERRORS.TRANSACTION.FEES_NOT_ENOUGH
+            );
+        });
+
+        it("Should fail: when unfunded account sends fund in group txn", function () {
+            assert.equal(interpreter.runtime.ctx.state.accounts
+                .get(johnAcc.address)?.balance()
+                , 0n);
+
+            const prog = `
+				itxn_begin
+				int pay
+				itxn_field TypeEnum
+				txn Applications 2
+				app_params_get AppAddress
+				assert
+				itxn_field Receiver
+				int 1000
+				itxn_field Amount
+				int 0
+				itxn_field Fee
+				itxn_next
+				int pay
+				itxn_field TypeEnum
+				txn Sender
+				itxn_field Receiver
+				int 1000
+				itxn_field Amount
+				int 2000
+				itxn_field Fee
+				int 1
+				return
+				`;
+            assert.doesNotThrow(() => executeTEAL(prog));
```

Comment:
this one should fail. assert.throws()

Reply:
or expect.runtimeError rather

Reply:
thanks for pointing out. I didn't submit the txn so it was passing. Fixed it.